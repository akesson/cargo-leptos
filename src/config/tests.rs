use camino::Utf8PathBuf;

use super::Config;

fn opts(project: Option<&str>) -> crate::Opts {
    crate::Opts {
        release: false,
        project: project.map(|s| s.to_string()),
        verbose: 0,
        bin_features: Vec::new(),
        lib_features: Vec::new(),
    }
}

#[test]
fn test_project() {
    let cli = opts(None);

    let conf = Config::load(cli, &Utf8PathBuf::from("examples/project/Cargo.toml"), true).unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace() {
    let cli = opts(None);

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project1() {
    let cli = opts(Some("project1"));

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project2() {
    let cli = opts(Some("project2"));

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}
