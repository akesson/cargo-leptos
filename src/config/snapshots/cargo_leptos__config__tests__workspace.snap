---
source: src/config/tests.rs
expression: conf
---
Config {
    projects: [
        Project {
            name: "project1",
            lib: LibPackage {
                name: "front-package",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/front_package.wasm",
                    dest: "target/site/project1/pkg/project1.wasm",
                    site: "pkg/project1.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/project1/pkg/project1.js",
                    site: "pkg/project1.js",
                },
                features: [
                    "hydrate",
                ],
                default_features: false,
                output_name: "project1",
                path_deps: "project1/app/src, project1/front/src",
                ..
            },
            bin: BinPackage {
                name: "server-package",
                exe_file: "target/server/debug/server-package",
                target: "server-package",
                features: [
                    "ssr",
                ],
                default_features: false,
                path_deps: "project1/app/src, project1/server/src",
                ..
            },
            style: Some(
                StyleConfig {
                    file: SourcedSiteFile {
                        source: "project1/css/main.scss",
                        dest: "target/site/project1/pkg/project1.css",
                        site: "pkg/project1.css",
                    },
                    browserquery: "defaults",
                },
            ),
            watch: true,
            release: false,
            site: Site {
                addr: 127.0.0.1:3000,
                reload: 127.0.0.1:3001,
                root_dir: "target/site/project1",
                pkg_dir: "pkg",
                file_reg: {},
                ext_file_reg: {},
            },
            end2end: None,
            assets: Some(
                AssetsConfig {
                    dir: "project1/assets",
                },
            ),
            ..
        },
        Project {
            name: "project2",
            lib: LibPackage {
                name: "project2",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/project2.wasm",
                    dest: "target/site/project2/pkg/project2.wasm",
                    site: "pkg/project2.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/project2/pkg/project2.js",
                    site: "pkg/project2.js",
                },
                features: [
                    "hydrate",
                ],
                default_features: false,
                output_name: "project2",
                path_deps: "project2/src",
                ..
            },
            bin: BinPackage {
                name: "project2",
                exe_file: "target/server/debug/project2",
                target: "project2",
                features: [
                    "ssr",
                ],
                default_features: false,
                path_deps: "project2/src",
                ..
            },
            style: Some(
                StyleConfig {
                    file: SourcedSiteFile {
                        source: "project2/src/main.scss",
                        dest: "target/site/project2/pkg/project2.css",
                        site: "pkg/project2.css",
                    },
                    browserquery: "defaults",
                },
            ),
            watch: true,
            release: false,
            site: Site {
                addr: 127.0.0.1:3000,
                reload: 127.0.0.1:3001,
                root_dir: "target/site/project2",
                pkg_dir: "pkg",
                file_reg: {},
                ext_file_reg: {},
            },
            end2end: None,
            assets: Some(
                AssetsConfig {
                    dir: "project2/src/assets",
                },
            ),
            ..
        },
    ],
    cli: Opts {
        release: false,
        project: None,
        lib_features: [],
        bin_features: [],
        verbose: 0,
    },
    watch: true,
    ..
}
